react/src/lib/components/SubsurfaceViewer/layers/triangle/webworker.ts

```diff
@@ -0,0 +1,166 @@
+
+        const v2_idx = indexArray[3 * t_no + 1];
+        vertexTrianglesArray[v2_idx].push(t_no);
+
+        const v3_idx = indexArray[3 * t_no + 2];
+        vertexTrianglesArray[v3_idx].push(t_no);
+    }
+
+    // The normal for each vertex is the mean of the normals belonging to the triangles that
+    // share this vertex.
+    for (let vertex_no = 0; vertex_no < nvertices; vertex_no++) {
+        const n_triangles = vertexTrianglesArray[vertex_no].length; // No triangles this vertex is part of. Will be at leat 1.
+        const t0 = vertexTrianglesArray[vertex_no][0]; // first triangle
+        const normal_mean = [
+            trianglesNormals[3 * t0 + 0],
+            trianglesNormals[3 * t0 + 1],
+            trianglesNormals[3 * t0 + 2],
+        ];
+        // possible rest of triangles
+        for (let t_no = 1; t_no < n_triangles; t_no++) {
+            const t = vertexTrianglesArray[vertex_no][t_no];
+            const normal = [
+                trianglesNormals[3 * t + 0],
+                trianglesNormals[3 * t + 1],
+                trianglesNormals[3 * t + 2],
+            ];
+
+            normal_mean[0] += normal[0];
+            normal_mean[1] += normal[1];
+            normal_mean[2] += normal[2];
+        }
+
+        normalize(normal_mean as Vec);
+
+        vertexsNormals[3 * vertex_no + 0] = normal_mean[0];
+        vertexsNormals[3 * vertex_no + 1] = normal_mean[1];
+        vertexsNormals[3 * vertex_no + 2] = normal_mean[2];
+
+        // KEEP!
+        // DEBUG SHOW NORMALS ////////////////////////////////
+        // const x0 = vertexArray[3 * vertex_no + 0];
+        // const y0 = vertexArray[3 * vertex_no + 1];
+        // const z0 = vertexArray[3 * vertex_no + 2];
+
+        // const scale = 5;
+        // const x1 = x0 + normal_mean[0] * scale;
+        // const y1 = y0 + normal_mean[1] * scale;
+        // const z1 = z0 + normal_mean[2] * scale;
+
+        // line_positions.push(x0, y0, z0);
+        // line_positions.push(x1, y1, z1);
+        ///////////////////////////////////////////////////////
+    }
+
+    const geometryTriangles: GeometryTriangles = {
+        drawMode: 4, // corresponds to GL.TRIANGLES,
+        attributes: {
+            positions: { value: new Float32Array(vertexArray), size: 3 },
+            normals: { value: new Float32Array(vertexsNormals), size: 3 },
+            vertex_indexs: { value: new Int32Array(indexArray), size: 1 },
+        },
+        vertexCount: indexArray.length,
+        indices: { value: new Uint32Array(indexArray), size: 1 },
+    };
+
+    const geometryLines: GeometryLines = {
+        drawMode: 1, // corresponds to GL.LINES,
+        attributes: {
+            positions: { value: new Float32Array(line_positions), size: 3 },
+        },
+        vertexCount: line_positions.length / 3,
+    };
+
+    //const t1 = performance.now();
+    // Keep this.
+    //console.log(`Task makeMesh took ${(t1 - t0) * 0.001}  seconds.`);
```

Comment:
Does this instrumentation reduce performance? If not we could consider not disabling it and use the `console.debug()` statement.

Reply:
Fixed